### ISSUE TITLE
Accessory never shown when alwaysVisible is false

### ISSUE THREAD

**Issue by TuurDutoit** [ardaogulcan/react-native-keyboard-accessory#42]
When `alwaysVisible` is set to false, the accessory never shows, not even when the keyboard comes up. I have narrowed down the issue to [this line](https://github.com/ardaogulcan/react-native-keyboard-accessory/blob/74e89afe94601674918f62d30206c81c43647743/src/KeyboardAccessoryView.js#L75). When the `keyboardWillShow` or `keyboardDidShow` event is received, [these lines](https://github.com/ardaogulcan/react-native-keyboard-accessory/blob/74e89afe94601674918f62d30206c81c43647743/src/KeyboardAccessoryView.js#L114-L118) set the right `accessoryHeight` in the state. When the children layout, however, the line mentioned previously resets this value to 0, even if `isKeyboardVisible` is true.

I fixed the issue by changing:
```js
accessoryHeight: this.props.alwaysVisible ? layoutEvent.nativeEvent.layout.height : 0,
```

to:
```js
accessoryHeight: this.props.alwaysVisible || this.state.isKeyboardVisible ? layoutEvent.nativeEvent.layout.height : 0,
```

**Comment by dinotrnka** [comment]
Hello,

Is there going to be a release with this fix?
It's pretty important and this PR fixed the issue for me too.

**Comment by dinotrnka** [comment]
Bumping this question so it doesn't get stale.

**Comment by dinotrnka** [comment]
Hi,

Is this library maintained or it's been abandoned?

**Comment by ardaogulcan** [comment]
Hi,

Library is not abandoned. Please reproduce this issue on expo snack so I can investigate.

**Comment by TuurDutoit** [comment]
This problem only occurs when the content of the accessory layouts *after* the keyboard is shown, because [these lines](https://github.com/ardaogulcan/react-native-keyboard-accessory/blob/74e89afe94601674918f62d30206c81c43647743/src/KeyboardAccessoryView.js#L114-L118) set the correct height for the accessory (they don't treat `alwaysVisible` differently), while [this line](https://github.com/ardaogulcan/react-native-keyboard-accessory/blob/74e89afe94601674918f62d30206c81c43647743/src/KeyboardAccessoryView.js#L75) doesn't (it sets the height to 0 if `alwaysVisible` is false). This works for simple cases where the accessory has a fixed size. However, if it needs to layout while the keyboard is open (e.g. when using a scrollview), things start to break.

As [this line](https://github.com/ardaogulcan/react-native-keyboard-accessory/blob/74e89afe94601674918f62d30206c81c43647743/src/KeyboardAccessoryView.js#L160) in the render method already checks whether the keyboard is visible or the accessory should always be shown, and sets the height accordingly, the same check shouldn't be carried out in `onChildrenLayout`. Doing so results in `render()` setting `height: 0`, even when `isKeyboardVisible` is true.

You can see the issue here: https://snack.expo.io/Sywe5VjvH
This simply prints out the content of the input field, with newlines between every character. It's a stupid example, but it makes the accessory adapt its height after the keyboard is shown. You can clearly see how it shows up fine at first (`handleKeyboardShow` is called after `onChildrenLayout`, setting the correct height), but as you start typing, the accessory disappears behind the keyboard (`onChildrenLayout` is called as the view has to adapt, setting `accessoryHeight: 0` because `alwaysVisible` is false).

**Comment by ardaogulcan** [comment]
@TuurDutoit Thanks for the detailed explanation and for PR. Your fix released with [v0.1.10](https://github.com/ardaogulcan/react-native-keyboard-accessory/releases/tag/0.1.10)
